Added unit test for MapboxTimer #2497

cafesilencio opened the pull request:
## Description
Issue 2497 added unit testing for MapboxTimer.

RingerJK commented on libnavigation-util/src/test/java/com/mapbox/navigation/utils/timer/MapboxTimerTest.kt:
```diff
@@ -0,0 +1,42 @@
+        timer.start()
+        delay(150)
+        timer.stop()
+        delay(150)
```
why do you need one more delay?

cafesilencio replied:
I"m testing the stop method in this test. If the last delay is removed and stop() is commented out the test will pass erroneously.  The delay is needed so that the implementation will tick one more time and the implementation's check of isActive should indicate that the timer is not active and therefore the lambda wouldn't get called an additional time.

RingerJK commented on libnavigation-util/src/test/java/com/mapbox/navigation/utils/timer/MapboxTimerTest.kt:
```diff
@@ -0,0 +1,42 @@
+import org.junit.Before
+import org.junit.Test
+
+class MapboxTimerTest {
```
suggest to add one more test when timer is canceled and lambda is not called

cafesilencio replied:
Good suggestion. I added it.

Guardiola31337 replied:
Saw this https://github.com/mapbox/mapbox-navigation-android/pull/2499/commits/f476b49cd1399344df56d3dde03ae03add41616e @cafesilencio and sounds good to me 👍

cafesilencio commented:
@abhishek1508 that's a good suggestion. I found that difficult to test given the way the job isn't visible outside the class.  As I understand the requirement only one coroutine should be created and launched  for a given instance of MapboxTimer regardless of the number of times the start method is called. That is, multiple calls to start() shouldn't create and launch multiple coroutines.  I modified the implementation to make the job variable immutable using Kotlin delegation. I added tests for the implementation as well. If there are objections to my changes to the implementation please let me know an I will put the original implementation back.
